code-challenges.test.js

```diff
@@ -14,46 +16,159 @@
+// use .map() method to iterate thopugh the array
+//return the array outputfrom the map action
+//access the first index of each value and convert it to uppercase using .toUpperCase() method
+//use + to concattenate the rest of the word
+
+// const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+// const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+
+// const capitalized = (array) => {
+//     return array.map((value) => {
+//         return value[0].toUpperCase() + value.substring(1)
+//     })
+
+// }
+
+//PASSED -> Test Suites: 1 passed, 1 total
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+//use a describe statement:
+// use a describe method that takes an argument of a sting
+// describe ("indexOfFirstVowel", () => {})
+//next, I use it statement which is nested inside describe block, to describe what the test is doing. 
+// it("takes in a string and logs the index of the first vowel", () => {})
+//next, I use expect statement to call the function and matcher method .toEqual.
+//.toEqual contains the expected output of the function. It compare the actual output to the output in the test
+//to run the tesy I use yarn jest command 
+
+describe("indexOfFirstVowel", () => {
+    it("takes in a string and logs the index of the first vowel", () => {
+        expect(indexOfFirstVowel(vowelTester1)).toEqual(1)
+        expect(indexOfFirstVowel(vowelTester2)).toEqual(0)
+        expect(indexOfFirstVowel(vowelTester3)).toEqual(2)
+
+    })
+})
+//FAIL ->  ReferenceError: indexOfFirstVowel is not defined
+
+// b) Create the function that makes the test pass.
+
+//declare a function called indexOfFirstVowel that takes a string as an argument
+// use for loop to iterate though the string and check the values of each index
+//use conditional statement if and OR operator to check if it contains a vowel, if so then return an index of the first vowel
+
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
-// b) Create the function that makes the test pass.
+const indexOfFirstVowel = (str) => {
+    for (i = 0; i < str.length; i++) {
+        if (
+            str[i] === "a" ||
+            str[i] === "e" ||
+            str[i] === "i" ||
+            str[i] === "o" ||
+            str[i] === "u"
+        ) {
+            return i
+        }
+    }
+}
+//PASS -> Test Suites: 1 passed, 1 total
+
+//I was also trying to figure out how to do it using .filter(). I believ I the string shoulb be converted to an array first. But then I was so confused and decided to use a fore loop and if statement
```

Comment:
This problem is definitely a bit easier to write out using a for loop.  To use higher-order functions, you will likely need to use more than just filter to accomplish this task.